helm/charts/hydra/templates/janitor-cron-job.yaml

```diff
@@ -0,0 +1,103 @@
+{{- if .Values.janitor.enabled -}}
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: {{ include "hydra.fullname" . }}-janitor
+  {{- if .Release.Namespace }}
+  namespace: {{ .Release.Namespace }}
+  {{- end }}
+  labels:
+    {{- include "hydra.janitor.labels" . | nindent 4 }}
+    {{- with .Values.janitor.cronjob.labels }}
+      {{- toYaml . | nindent 4 }}
+    {{- end }}
+  annotations:
+    {{- with .Values.janitor.cronjob.annotations }}
+      {{- toYaml . | nindent 4 }}
+    {{- end }}
+spec:
+  concurrencyPolicy: Forbid
+  schedule: {{ .Values.janitor.cronjob.schedule | quote }}
+  jobTemplate:
+    spec:
+      template:
+        metadata:
+          labels:
+            {{- include "hydra.janitor.labels" . | nindent 12 }}
+            {{- with .Values.janitor.cronjob.labels }}
+              {{- toYaml . | nindent 12 }}
+            {{- end }}
+            {{- with .Values.janitor.cronjob.podMetadata.labels }}
+              {{- toYaml . | nindent 12 }}
+            {{- end }}
+          annotations:
+            {{- include "hydra.annotations.checksum" . | nindent 12 -}}
+            {{- with .Values.janitor.cronjob.annotations }}
+              {{- toYaml . | nindent 12 }}
+            {{- end }}
+            {{- with $.Values.janitor.cronjob.podMetadata.annotations }}
+              {{- toYaml . | nindent 12 }}
+            {{- end }}
+        spec:
+          restartPolicy: OnFailure
+          {{- with .Values.imagePullSecrets }}
+          imagePullSecrets:
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+          volumes:
+            - name: {{ include "hydra.name" . }}-config-volume
+              configMap:
+                name: {{ include "hydra.fullname" . }}
+          containers:
+            - name: janitor
+              image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+              imagePullPolicy: {{ .Values.image.pullPolicy }}
+              {{- if .Values.janitor.cronjob.securityContext.enabled }}
+              securityContext:
+                {{- omit .Values.janitor.cronjob.securityContext "enabled" | toYaml | nindent 16 }}
+              {{- end }}
+              command: ["hydra"]
+              args:
```

Comment:
I'm not sure if this is overcomplicated. Wouldn't it be simpler to set default args in values yaml following the formula:

```yaml
janitor:
  # -- Default args send to the hydra janitor command. You can override this field to input your own arguments
  args:
    - janitor
    - --config
    - /etc/config/hydra.yaml
    - --read-from-env
    - $DSN
  # -- Extra Arguments appended to the hydra janitor command
  extraArgs: []
```

Reply:
I wanted to structure the chart values and avoid leaving this freedom to the user in terms of command line customization. However, if you think it is too rigid a thought, I have nothing negative about changing my mind.

Reply:
I am not against this structure, but think we should leave the possibility for customisation :) How about leaving the structure, which most probably will be enough for most, but for power users allow a `customArgs` which would override the whole array. What do you think?

Reply:
Considering that `customArgs` is already provided for the `Job` that runs the migrations, it makes sense 😄